src/fiatconnect/TransferStatusScreen.test.tsx

```diff
@@ -0,0 +1,124 @@
+import { CICOFlow } from 'src/fiatExchanges/utils'
+import { FiatConnectQuoteSuccess } from 'src/fiatconnect'
+import TransferStatusScreen from 'src/fiatconnect/TransferStatusScreen'
+import { navigate, navigateBack, navigateHome } from 'src/navigator/NavigationService'
+import React from 'react'
+import networkConfig from 'src/web3/networkConfig'
+
+describe('TransferStatusScreen', () => {
+  const mockStore = (overrides: any = {}) => {
+    const store = createMockStore({
+      fiatConnect: {
+        transfer: mockFiatConnectTransfers[0],
+        ...overrides,
+      },
+    })
+    store.dispatch = jest.fn()
+    return store
+  }
+
+  const mockQuote = new FiatConnectQuote({
+    flow: CICOFlow.CashOut,
+    fiatAccountType: FiatAccountType.BankAccount,
+    quote: mockFiatConnectQuotes[1] as FiatConnectQuoteSuccess,
+  })
+
+  const mockScreenProps = () =>
+    getMockStackScreenProps(Screens.FiatConnectTransferStatus, {
+      flow: CICOFlow.CashOut,
+      fiatAccount: {
+        fiatAccountId: 'some-fiat-account-id',
+        accountName: 'some-friendly-name',
+        institutionName: 'some-bank',
+        fiatAccountType: FiatAccountType.BankAccount,
+      },
+      normalizedQuote: mockQuote,
+    })
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('success view', () => {
+    it('sets header options', () => {
+      const store = mockStore()
+      const mockProps = mockScreenProps()
+      render(
+        <Provider store={store}>
+          <TransferStatusScreen {...mockProps} />
+        </Provider>
+      )
+      expect(mockProps.navigation.setOptions).toBeCalled()
+    })
+    it('navigates home on success', () => {
+      const store = mockStore()
+      const { queryByTestId, getByTestId } = render(
+        <Provider store={store}>
+          <TransferStatusScreen {...mockScreenProps()} />
+        </Provider>
+      )
+      expect(queryByTestId('Continue')).toBeTruthy()
+      fireEvent.press(getByTestId('Continue'))
+      expect(navigateHome).toHaveBeenCalledWith()
+    })
+
+    it('shows TX details on Celo Explorer on success', () => {
+      const store = mockStore()
+      const { queryByTestId, getByTestId } = render(
+        <Provider store={store}>
+          <TransferStatusScreen {...mockScreenProps()} />
+        </Provider>
+      )
+      expect(queryByTestId('txDetails')).toBeTruthy()
+      fireEvent.press(getByTestId('txDetails'))
+      expect(navigate).toHaveBeenCalledWith(Screens.WebViewScreen, {
+        uri: `${networkConfig.celoExplorerBaseTxUrl}${mockFiatConnectTransfers[0].txHash}`,
+      })
+    })
+  })
+
+  describe('failure view', () => {
+    it('sets header options', () => {
+      const store = mockStore()
+      const mockProps = mockScreenProps()
+      render(
+        <Provider store={store}>
+          <TransferStatusScreen {...mockProps} />
+        </Provider>
+      )
+      expect(mockProps.navigation.setOptions).toBeCalled()
```

Comment:
could also use partial matchers (expect.objectContaining, expect.any, etc) if you wanted to check for specific things set.
```js
expect(mockProps.navigation.setOptions).toBeCalledWith(
  expect.objectContaining({ headerLeft: expect.any(Function) })
)
```
or could also mock out emptyHeaders and do a full object match.